src/components/builder.ts

```diff
@@ -168,20 +167,20 @@ export class Builder {
         return true
     }
 
+    isAutoBuildEnabled() {
+        return !this.saving
+    }
+
     async saveActive() {
-        await this.extension.removeBuildOnSaveEvent(this.buildOnSaveEvent)
-        this.extension.manager.unregisterbuildOnFileChanged()
+        this.saving = true
         await vscode.window.activeTextEditor?.document.save()
-        this.buildOnSaveEvent = this.extension.createBuildOnSaveEvent()
-        this.extension.manager.registerbuildOnFileChanged()
+        setTimeout(() => this.saving = false, 500)
     }
 
     private async saveAll() {
-        await this.extension.removeBuildOnSaveEvent(this.buildOnSaveEvent)
-        this.extension.manager.unregisterbuildOnFileChanged()
+        this.saving = true
         await vscode.workspace.saveAll()
-        this.buildOnSaveEvent = this.extension.createBuildOnSaveEvent()
-        this.extension.manager.registerbuildOnFileChanged()
+        setTimeout(() => this.saving = false, 500)
```

Comment:
Maybe we could use `latex-workshop.latex.autoBuild.interval` instead of `500` here.

Reply:
I agree. The new implementation should make the whole logic more clear.